src/NUnitTestExtractor/NUnitTestExtractor/Program.cs

```diff
@@ -55,44 +57,78 @@ static void Main(string[] args)
+        private static void GetTests(IList<string> dlls, Level level)
+        {
+            foreach(string dll in dlls)
+            {
+                Assembly assembly = null;
+                try
                 {
-                    writer = new StreamWriter(Console.OpenStandardOutput());
-                }
-                else
+                    assembly = Assembly.LoadFrom(dll);
+                }catch(FileNotFoundException e)
                 {
-                    writer = new StreamWriter(outputDirectory + name +".txt", true);
+                    Console.Error.WriteLine("{0}: file {1} was not found on system", e.GetType(), dll);
+                    continue;
                 }
-                foreach (string dll in dllDirectories)
+               
+                foreach (Type type in assembly.GetTypes())
                 {
-                    if (File.Exists(dll))
-                    {
-                        writer.Write("{0}|{1}", dll, level + Environment.NewLine);
-                    }
-                    else
+                    foreach (MethodInfo methodInfo in type.GetMethods())
                     {
-                        Console.Error.WriteLine("Error: Unable to append {0} as it cannot be found on the system", dll);
+                        var attributes = methodInfo.GetCustomAttributes(true);
+                        foreach (var attr in attributes)
+                        {
+                            if (attr is NUnit.Framework.TestAttribute)
+                            {                               
+                                string value = "";
+                                
+                                switch (level)
+                                {
+                                    case Level.Namespace:
+                                        value = type.Namespace;
+                                        break;
+                                    case Level.Class:
+                                        value = methodInfo.DeclaringType.ToString();
+                                        break;
+                                    case Level.Function:
+                                        string method = methodInfo.ToString().Substring(methodInfo.ToString().IndexOf(" ", StringComparison.OrdinalIgnoreCase) + 1);
+                                        value = methodInfo.DeclaringType + "." + method;
+                                        break;
+                                }
+                                StreamWriter writer;
+                                if (_options.Output== null)
+                                {
+                                    writer = new StreamWriter(Console.OpenStandardOutput());
+                                }
+                                else
+                                {
+                                    writer = new StreamWriter(_options.Output+ "file.txt", true);
```

Comment:
Remove the "file.txt".  _options.Output should already have the right filename to write to.